server/podcasts/podcasts.go

```diff
@@ -170,26 +171,73 @@ func (p *Podcasts) AddEpisode(podcastID int, item *gofeed.Item) error {
 		duration = getSecondsFromString(item.ITunesExt.Duration)
 	}
 
+	if episode, ok := p.findEnclosureAudio(podcastID, duration, item); ok {
+		if err := p.DB.Save(episode).Error; err != nil {
+			return err
+		}
+		return nil
+	}
+	if episode, ok := p.findMediaAudio(podcastID, duration, item); ok {
+		if err := p.DB.Save(episode).Error; err != nil {
+			return err
+		}
+		return nil
+	}
+	// hopefully shouldnt reach here
+	log.Println("failed to find audio in feed item, skipping")
+	return nil
+}
+
+func isAudio(mediaType, url string) bool {
+	if mediaType != "" && strings.HasPrefix(mediaType, "audio") {
+		return true
+	}
+	switch filepath.Ext(url) {
+	case ".flac", ".mp3", ".ogg", ".opus", ".wav", ".alac":
+		return true
+	}
+	return false
+}
```

Comment:
do you think it could be possible to leverage `server/mime.FromExtension` here? it can return a bool
https://github.com/sentriz/gonic/blob/master/server/mime/mime.go
feel free to update it too if it could work

Reply:
:+1: